### PR TITLE
Backport Scala 2.13 BatchingExecutor into Akka

### PR DESCRIPTION
**Pull request by johanandren**
Refs #26383

Two tests in ExecutionContextSpec still failing

### CODE REVIEW

**Comment by johanandren** [comment]
Two tests currently failing: 

 * “must be able to avoid starvation when Batching is used and Await/blocking is called”
  Causing a stack overflor looping on `akka.dispatch.BatchingExecutor$AsyncBatch.blockOn(BatchingExecutor.scala:264)`
 * “must work with same-thread dispatcher as executor with blocking”

File: akka-actor-tests/src/test/scala/akka/dispatch/ExecutionContextSpec.scala [hunk 1]
```diff
@@ -46,8 +46,7 @@ class ExecutionContextSpec extends AkkaSpec with DefaultTimeout {
       import system.dispatcher
 
       def batchable[T](f: => T)(implicit ec: ExecutionContext): Unit =
-        ec.execute(new Batchable {
-          override def isBatchable = true
+        ec.execute(new Runnable with Batchable {
```
Comment (hepin1989) [inline akka-actor-tests/src/test/scala/akka/dispatch/ExecutionContextSpec.scala]: make it a dedicated class?

Comment (hepin1989) [reply akka-actor-tests/src/test/scala/akka/dispatch/ExecutionContextSpec.scala]: RunBatch

**Comment by johanandren** [comment]
Let's close this for now and possibly pick it up in the future.

**Pull request closed without merging**